sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -230,25 +294,67 @@ object RmmRapidsRetryIterator extends Arm {
    * @param splitPolicy a function that can split an item of type T into a Seq[T]. The split
    *                    function must close the item passed to it.
    */
-  class RmmRapidsRetryAutoCloseableIterator[T <: AutoCloseable, K](
+  class AutoCloseableAttemptSpliterator[T <: AutoCloseable, K](
       input: Iterator[T],
       fn: T => K,
       splitPolicy: T => Seq[T])
-      extends RmmRapidsRetryIterator(input, fn, splitPolicy)
-        with Arm {
-
-    def this(
-        input: Iterator[T],
-        fn: T => K) = {
+      extends Iterator[K]
```

Comment:
Iterator[K] is not needed if we also extend Spliterator[K]

Reply:
done